### PR TITLE
ML Assisted AprilTag Pipeline Settings

### PR DESCRIPTION
**Pull request by DoctorFogarty**
Add NPU-accelerated, AI-assisted AprilTag detection to PhotonVision 
Screenshot 2026-01-18 204036

Stage 1 (NPU): YOLO model detects AprilTag bounding boxes via neural network inference
Stage 2 (CPU): WPILib AprilTag detector runs only within detected ROIs for accurate detection and ID decoding

This leverages existing ML infrastructure for supported NPU platforms:

RK3588 (Orange Pi 5, Rock 5C, CoolPi 4B) - RKNN backend (Should work, but untested)
QCS6490 (Rubik Pi 3) - TFLite backend (Tested)

Toggle in AprilTag pipeline settings: traditional vs AI-assisted detection
Graceful fallback to traditional detection when ML is unavailable or finds no tags. 
Maintains full pose estimation accuracy (corner and homography transformed)
Platform auto-detection enables ML only on supported hardware

AprilTagROIDetectionPipe - Runs YOLO inference on NPU to find tag bounding boxes
AprilTagROIDecodePipe - Extracts ROI submats, runs traditional detector, maps coordinates back to full frame
Existing pose estimation pipes unchanged 

Files Changed
AprilTagPipelineSettings.java - Add ML settings (useMLDetection, confidence, NMS, ROI expansion)
AprilTagPipeline.java - Add hybrid detection path with fallback logic
AprilTagROIDetectionPipe.java - NEW: ML inference pipe
AprilTagROIDecodePipe.java - NEW: ROI decode + coordinate mapping pipe
NeuralNetworkModelManager.java - Add model lookup methods
PipelineTypes.ts - Add TypeScript types for ML settings
AprilTagTab.vue - Add UI controls for ML detection toggle

Model
YOLOv8n trained on AprilTag dataset (Roboflow)
https://app.roboflow.com/johnspace-ivyq6/apriltagimages-mijky/1/export
This is meant to be a starting point for a proof of concept not the best dataset possible.
INT8 quantized, 640x640 input
Converted via Qualcomm AI Hub for TFLite format

Merge checklist:
- [ ] Pull Request title is [short, imperative summary](https://cbea.ms/git-commit/) of proposed changes
- [ ] The description documents the _what_ and _why_
- [ ] If this PR changes behavior or adds a feature, user documentation is updated
- [ ] If this PR touches photon-serde, all messages have been regenerated and hashes have not changed unexpectedly
- [ ] If this PR touches configuration, this is backwards compatible with settings back to v2025.3.2
- [ ] If this PR touches pipeline settings or anything related to data exchange, the frontend typing is updated
- [ ] If this PR addresses a bug, a regression test for it is added

### CODE REVIEW

File: apriltagmodel/yolov8_det-tflite-w8a8/apriltag-640-640-yolov8n.tflite
Comment (samfreund) [inline apriltagmodel/yolov8_det-tflite-w8a8/apriltag-640-640-yolov8n.tflite:1]: We don't need this model twice

File: apriltagmodel/yolov8_det-tflite-w8a8/tool-versions.yaml
Comment (samfreund) [inline apriltagmodel/yolov8_det-tflite-w8a8/tool-versions.yaml:1]: delete file

File: photon-client/src/components/dashboard/tabs/AprilTagTab.vue [hunk 1]
```diff
@@ -18,6 +19,12 @@ const { mdAndDown } = useDisplay();
 const interactiveCols = computed(() =>
   mdAndDown.value && (!useStateStore().sidebarFolded || useCameraSettingsStore().isDriverMode) ? 8 : 7
 );
+
+// Check if ML detection is available on this platform
+const mlDetectionAvailable = computed(() => {
+  const supportedBackends = useSettingsStore().general.supportedBackends || [];
+  return supportedBackends.length > 0;
```
Comment (samfreund) [inline photon-client/src/components/dashboard/tabs/AprilTagTab.vue]: This can be one line.

File: apriltagmodel/yolov8_det-tflite-w8a8/apriltag-640-640-yolov8n.tflite
Comment (TheJohnFogarty) [reply apriltagmodel/yolov8_det-tflite-w8a8/apriltag-640-640-yolov8n.tflite:1]: you'd rather I leave the .pt file?

Comment (samfreund) [reply apriltagmodel/yolov8_det-tflite-w8a8/apriltag-640-640-yolov8n.tflite:1]: The PyTorch file is useful for conversion to other formats, the tflite file is already in photon-server if I'm not mistaken.

File: photon-core/src/main/java/org/photonvision/vision/pipe/impl/AprilTagROIDecodePipe.java [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+        double centerX = det.getCenterX() + roiOffset.x;
+        double centerY = det.getCenterY() + roiOffset.y;
+
+        // Homography is passed through unchanged - it's not used for pose estimation
+        return new AprilTagDetection(
+                det.getFamily(),
+                det.getId(),
+                det.getHamming(),
+                det.getDecisionMargin(),
+                det.getHomography(), // Unchanged - not used for pose estimation
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/vision/pipe/impl/AprilTagROIDecodePipe.java]: This comment is incorrect -- this is used for pose est by the umich pose estimator

File: photon-core/src/main/java/org/photonvision/vision/pipe/impl/AprilTagROIDecodePipe.java [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+import org.photonvision.vision.pipe.CVPipe;
+
+/**
+ * Pipe that decodes AprilTags within ROIs detected by ML. For each ROI from ML detection, it
+ * extracts a sub-image, runs the traditional WPILib AprilTag detector, and maps the detected corner
+ * coordinates back to full-frame coordinates.
+ *
+ * <p>CRITICAL: This class contains the coordinate mapping logic. The homography does NOT need
+ * transformation - only corner and center coordinates need mapping. Pose estimation in
+ * AprilTagPoseEstimatorPipe uses corners exclusively.
```
Comment (Gold856) [inline photon-core/src/main/java/org/photonvision/vision/pipe/impl/AprilTagROIDecodePipe.java]: This is not true. AprilTagPoseEstimator does use homography.

**Review by samfreund — CHANGES_REQUESTED** [review]

File: photon-client/src/components/dashboard/tabs/AprilTagTab.vue (hunk 1 above)
Comment (samfreund) [reply photon-client/src/components/dashboard/tabs/AprilTagTab.vue]: Look at how we do it in other places
